### PR TITLE
Add libtasn1/4.16.0 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libtasn1/4.16.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x]  I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libtasn1/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        conf_args = [
+            "--disable-doc",
+        ]
+        if self.options.shared:
+            conf_args.extend(["--enable-shared", "--disable-static"])
+        else:
+            conf_args.extend(["--disable-shared", "--enable-static"])
+        if "fPIC" in self.options.fields:
+            conf_args.append("--with-pic" if self.options.fPIC else "--without-pic")
```
Comment (madebr) [inline recipes/libtasn1/all/conanfile.py:61]:
```suggestion
```

**Comment by uilianries** [comment]
@madebr the recipe looks great, but I think the test package could be simplified. Both asn and cpp file are really big.

File: recipes/libtasn1/all/test_package/pkix.asn [hunk 2]
```diff
@@ -0,0 +1,954 @@
+-- Copyright (C) 2002-2014 Free Software Foundation, Inc.
```
Comment (uilianries) [inline recipes/libtasn1/all/test_package/pkix.asn:1]: Is it possible to reduce this file? Almost 1k lines, only for a simple test ...

Comment (SSE4) [reply recipes/libtasn1/all/test_package/pkix.asn:1]: it's not source code, but a test file. I think it's okay (we e.g. have test jpeg files in repo)

**Pull request merged**